pkg/controller/test/context_builder.go

```diff
@@ -111,6 +113,29 @@ func (b *Builder) Init() {
 	b.Client = kubefake.NewSimpleClientset(b.KubeObjects...)
 	b.CMClient = cmfake.NewSimpleClientset(b.CertManagerObjects...)
 	b.GWClient = gwfake.NewSimpleClientset(b.GWObjects...)
+	b.DiscoveryClient = discoveryfake.NewDiscovery().WithServerResourcesForGroupVersion(func(groupVersion string) (*metav1.APIResourceList, error) {
+		if groupVersion == networkingv1.SchemeGroupVersion.String() {
+			return &metav1.APIResourceList{
+				TypeMeta:     metav1.TypeMeta{},
+				GroupVersion: networkingv1.SchemeGroupVersion.String(),
+				APIResources: []metav1.APIResource{
+					{
+						Name:               "Ingresses",
```

Comment:
nit ([source](https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/restmapper/discovery_test.go#L49))
```suggestion
						Name:               "ingresses",
```